packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -123,6 +126,79 @@ void _debugRecordDownStream(String channelTypeName, String name,
   _debugLaunchProfilePlatformChannels();
 }
 
+class BackgroundIsolateBinaryMessenger extends BinaryMessenger {
+  final ReceivePort receivePort = ReceivePort();
+  final Map<int, Completer<ByteData?>> _completers = <int, Completer<ByteData?>>{};
+  int _messageCount = 0;
+
+  static BinaryMessenger? _backgroundIsolateBinaryMessenger;
+
+  static void ensureInitialized(ui.RootIsolateToken token) {
+    if (_backgroundIsolateBinaryMessenger == null) {
+      ui.PlatformDispatcher.instance.registerBackgroundIsolate(token);
+      final BackgroundIsolateBinaryMessenger portBinaryMessenger = BackgroundIsolateBinaryMessenger();
+      _backgroundIsolateBinaryMessenger = portBinaryMessenger;
+      portBinaryMessenger.receivePort.listen((dynamic message) {
+        try {
+          final List<dynamic> args = message as List<dynamic>;
+          final int identifier = args[0] as int;
+          final Uint8List bytes = args[1] as Uint8List;
+          final ByteData byteData = ByteData.sublistView(bytes);
+          final Completer<ByteData?> completer =
+              portBinaryMessenger._completers[identifier]!;
+          portBinaryMessenger._completers.remove(identifier);
+          completer.complete(byteData);
+        } catch (exception, stack) {
+          FlutterError.reportError(FlutterErrorDetails(
+            exception: exception,
+            stack: stack,
+            library: 'services library',
+            context:
+                ErrorDescription('during a platform message response callback'),
+          ));
+        }
+      });
+    }
+  }
+
+  static BinaryMessenger? get instance => _backgroundIsolateBinaryMessenger;
+
+  @override
+  Future<void> handlePlatformMessage(String channel, ByteData? data, ui.PlatformMessageResponseCallback? callback) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<ByteData?>? send(String channel, ByteData? message) {
+    final Completer<ByteData?> completer = Completer<ByteData?>();
+    final int messageIdentifier = ++_messageCount;
```

Comment:
trivial nit: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer--over-

Reply:
done